experiments/book-surprisal/ocr_surprisal.py

```
@@ -0,0 +1,90 @@
+    fp=pg["first_pass"]; sv=sorted(fp["surprisal"]); pct=sv[int(0.95*len(sv))]
+    spans=[m.span() for m in re.finditer(r"[A-Za-z]{3,}", fp["text"])]
+    for sp in random.sample(spans, min(3,len(spans))):
+        h=span_hit(sp, fp["offsets"], fp["surprisal"], pct)
+        if h is not None: rnd_hits.append(h)
+    # the fairer null: real words occurring exactly once in the whole refined
+    # book, legitimate but rare, up to three occurrences a page
+    rare=[m.span() for m in re.finditer(r"[A-Za-z]{4,}", fp["text"]) if freq.get(m.group(0))==1][:3]
+    for sp in rare:
+        h=span_hit(sp, fp["offsets"], fp["surprisal"], pct)
+        if h is not None: rare_hits.append(h)
+res["artifact_words_found"]=arts; res["artifact_hit_rate_at_page_p95"]=round(sum(art_hits)/max(1,len(art_hits)),3); res["artifact_occurrences"]=len(art_hits)
```

Ruff (0.16.3)

[error] Multiple statements on one line (semicolon)

(E702)